dubbo-admin-server/src/main/java/org/apache/dubbo/admin/service/SideService.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dubbo.admin.service;
+
+import org.apache.dubbo.admin.common.exception.ParamValidationException;
+import org.apache.dubbo.admin.common.util.Constants;
+import org.apache.dubbo.admin.common.util.SyncUtils;
+import org.apache.dubbo.common.URL;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentMap;
+
+public interface SideService {
```

Comment:
Because `consumerService `also needs to follow the same logic of `providerService#findVersionInApplication` (when `providerService `does not match the application name, it needs to be matched again to `consumerService`), so I extracted the code of `providerSerivice `into this interface, so that `consumerService `and `providerService `are shared.

As for the interface name, I didn't think of a better one, if I have a good suggestion, I can modify it.